install.sh

```diff
@@ -0,0 +1,157 @@
+
+    if [ ! -f .mvn/extensions.xml ]; then
+        echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+            <extensions>
+              ${EXTENSION}
+            </extensions>" > .mvn/extensions.xml
+    else
+        EXTENSION_REGISTERED=$(cat .mvn/extensions.xml | grep 'org.arquillian.smart.testing' -A 2 |  grep -oPm1 "(?<=<version>)[^<]+");
+
+        if [ ! ${EXTENSION_REGISTERED} ]; then
+            EXTENSION=$(echo ${EXTENSION} | sed -e "s#/#\\\/#g");
+            sed -i -E 's/(.*<extensions>)(.*)/\1\n'$EXTENSION'\2/g' .mvn/extensions.xml
+            echo "Installed Smart Testing Extension ${VERSION}"
+        else
+            echo -e "Smart Testing Extension already registered with version ${EXTENSION_REGISTERED}\c"
+
+            if [ ${INSTALL_SPECIFIC_VERSION} -eq 1 ]; then
+                if [ "${EXTENSION_REGISTERED}" != "${VERSION}" ]; then
+                    echo -e " - overwriting with ${VERSION}."
+                    override_version $VERSION
+                    echo -e "Updated Smart Testing Extension to ${VERSION}\c"
+                fi
+                echo "."
+            elif [ $EXTENSION_REGISTERED != $LATEST ]; then
+                read -r -p ". Do you want to override with latest ${LATEST}? [y/N] " response
+                case "$response" in
+                    [yY][eE][sS]|[yY])
+                          override_version ${LATEST}
+                          echo "Updated Smart Testing Extension to ${LATEST}"
+                        ;;
+                    *)
+
+                        ;;
+                esac
+            else
+                echo " which is the latest stable version."
+            fi
+        fi
+    fi
+
+    mv .mvn/extensions.xml .mvn/extensions-unformatted.xml
+    xmllint --format .mvn/extensions-unformatted.xml > .mvn/extensions.xml
+    rm .mvn/extensions-unformatted.xml
+}
+
+function override_version() {
+    cat .mvn/extensions.xml | awk -v groupId="org.arquillian.smart.testing" -v version=$1 -v RS="</extension>" '
+      $0 ~ "<groupId>" groupId "</groupId>" {
+        sub("<version>.*</version>", "<version>" version "</version>")
+      }
+      {printf "%s", $0 RS}' | sed '$ d' > .mvn/extensions-new.xml
+    mv .mvn/extensions-new.xml .mvn/extensions.xml
+}
+
+function ignore_smart_testing_artifacts() {
+    read -r -p "Do you want to add Smart Testing execution artifacts to .gitignore? [Y/n] " response
+        case "$response" in
+            [nN][oO]|[nN])
+                ;;
+            *)
+                if [ ! -f .gitignore ]; then
+                    touch ./.gitignore
+                fi
+                cat .gitignore | grep -q '.smart-testing' && EXISTS=1 || EXISTS=0
+                if [ ${EXISTS} == 0 ]; then
+                    echo "# Smart Testing Exclusions\n.smart-testing" >> ./.gitignore
```

Comment:
the resulting string in .gitignore is: `# Smart Testing Exclusions\n.smart-testing`
the command should be: `echo -e "# Smart Testing Exclusions\n.smart-testing" >> ./.gitignore`

Reply:
Strange. I will check again.